src/components/NotificationCenter/index.js

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import { values } from "lodash";
+import { connect } from "react-redux";
+import Notification from "./Notification";
+import TodoList from "../TodoList";
+
+
+export function NotificationCenter ({notifications, dismissNotification, deactivateNotification}) {
+  return (
+    <div className="notification-center">
+      <ul>
+        {notifications.map(notification => {
+          return (
+            <li key={notification.id}>
+              <Notification
+                notification={notification}
+                onDismiss={() => dismissNotification(notification.id)}
+                onDeactivate={() => deactivateNotification(notification.id, notification.todo.id)}
+              />
+            </li>
+          )
+        })}
+      </ul>
+    </div>
+  )
+}
+
+const mapStateToProps = (state, ownProps) => {
+  return {
+    notifications: values(state.notifications).filter((notification) => notification.active)
+  }
+}
+
+const mapDispatchToProps = (dispatch, ownProps) => {
+  return {
+    dismissNotification: (id) => {
+      dispatch({
+        type:'DISMISS_NOTIFICATION',
+        payload: {
+          id
+        }
+      })
+    },
+
+    deactivateNotification: (id, todoId, todoListId, updatedKeysInTodo) => {
+      dispatch({
+        type: 'DISMISS_NOTIFICATION',
+        payload: {
+          id
+        }
+      });
+
+      dispatch({
+        type: 'UPDATE_TODO',
+        payload: {
+          todoId,
+          updatedKeysInTodo,
+          todoListId: ownProps.id,
+          active: false
```

Comment:
If you carefully read the code that we used earlier when dispatching the action `'UPDATE_TODO'`, you'll notice that `updatedKeysInTodo` is a key of the payload.
If you want to deactivate a todo, you must do the following instead:
```javascript
dispatch({
        type: 'UPDATE_TODO',
        payload: {
          todoListId,
          todoId,
          updatedKeysInTodo: { active: false },
})
```

`updatedKeysInTodo` is the key in the payload whose value contains the changes that should be applied to the todo. In you case, passing `active: false` directly in the payload won't work. To be sure of that, you can go back to the todoList reducer and check how the `UPDATE_TODO` action is handled:
```javascript
case 'UPDATE_TODO':
      todoId = action.payload.todoId
      todoListId = action.payload.todoListId
      const updatedKeysInTodo = action.payload.updatedKeysInTodo

      todoList = state[todoListId]
      todos = todoList.todos
      todo = todos[todoId]

      return {
        ...state,
        [todoListId]: {
          ...todoList,
          todos: {
            ...todos,
            [todoId]: {
              ...todo,
              ...updatedKeysInTodo
            }
          }
        }
      }
```

It looks complicated, but the part that really interest us is the following:
```javascript
[todoId]: {
    ...todo,
    ...updatedKeysInTodo
 }
```
As you can maybe see, we are updating the value associated to the key `[todoId]` (let's say todoId=2) with a new object. To build this object, we first spread the current version of the todo (we copy all its keys and values), and then we spread the updates that we want to apply.

Reply:
So we can deduce from this code that `updatedKeysInTodo` should be an object containing all the keys that should change, and the values they should change to.